t/smoke/tests/quorum_queues/test_native_delayed_delivery.py

```diff
@@ -0,0 +1,169 @@
+
+            assert queue_arguments["x-message-ttl"] == pow(2, queue_level) * 1000
+
+            conf = default_worker_app.conf
+            assert queue_arguments["x-queue-type"] == conf.broker_native_delayed_delivery_queue_type
+
+    def test_native_delayed_delivery_exchange_configuration(self, celery_setup: CeleryTestSetup):
+        broker: RabbitMQManagementBroker = celery_setup.broker
+        api = broker.get_management_url() + "/api/exchanges"
+        response = requests.get(api, auth=HTTPBasicAuth("guest", "guest"))
+        assert response.status_code == 200
+        res = response.json()
+        assert isinstance(res, list)
+
+        res = [exchange for exchange in res if exchange["name"].startswith('celery_delayed')]
+
+        assert len(res) == 29
+
+        for exchange in res:
+            assert exchange["type"] == "topic"
+
+
+class test_broker_configuration_classic:
+    @pytest.fixture
+    def default_worker_app(self, default_worker_app: Celery) -> Celery:
+        app = default_worker_app
+        app.conf.broker_transport_options = {"confirm_publish": True}
+        app.conf.task_default_queue_type = "quorum"
+        app.conf.broker_native_delayed_delivery = True
+        app.conf.broker_native_delayed_delivery_queue_type = 'classic'
+        app.conf.task_default_exchange_type = 'topic'
+        app.conf.task_default_routing_key = 'celery'
+
+        return app
+
+    def test_native_delayed_delivery_queue_configuration(
+        self,
+        celery_setup: CeleryTestSetup,
+        default_worker_app: Celery
+    ):
+        broker: RabbitMQManagementBroker = celery_setup.broker
+        api = broker.get_management_url() + "/api/queues"
+        response = requests.get(api, auth=HTTPBasicAuth("guest", "guest"))
+        assert response.status_code == 200
+        res = response.json()
+        assert isinstance(res, list)
+
+        res = [queue for queue in res if queue["name"].startswith('celery_delayed')]
+
+        assert len(res) == 28
+
+        for queue in res:
+            queue_level = int(queue["name"].split("_")[-1])
+
+            queue_arguments = queue["arguments"]
+            if queue_level == 0:
+                assert queue_arguments["x-dead-letter-exchange"] == "celery_delayed_delivery"
+            else:
+                assert queue_arguments["x-dead-letter-exchange"] == f"celery_delayed_{queue_level - 1}"
+
+            assert queue_arguments["x-message-ttl"] == pow(2, queue_level) * 1000
+            assert queue_arguments["x-queue-type"] == 'classic'
+
+    def test_native_delayed_delivery_exchange_configuration(self, celery_setup: CeleryTestSetup):
+        broker: RabbitMQManagementBroker = celery_setup.broker
+        api = broker.get_management_url() + "/api/exchanges"
+        response = requests.get(api, auth=HTTPBasicAuth("guest", "guest"))
+        assert response.status_code == 200
+        res = response.json()
+        assert isinstance(res, list)
```

Comment:
Same

Reply:
I tried moving this to a fixture but I got an error due to a bug in pytest-celery.
I think this doesn't matter much.